packages/website-backend/src/api/RealTimeReportsController.ts

```diff
@@ -0,0 +1,231 @@
+import {
+  BadRequest,
+  InternalServerError,
+  NotFound,
+  Unauthorized,
+} from 'ts-httpexceptions';
+import MutationtEventServerOrchestrator from '../services/real-time/MutationtEventServerOrchestrator.js';
+import { MutationTestResult } from 'mutation-testing-report-schema';
+import { parseSlug } from './util.js';
+import { ReportValidator } from '../services/ReportValidator.js';
+import { PutReportResponse } from '@stryker-mutator/dashboard-contract';
+import Configuration from '../services/Configuration.js';
+
+const API_KEY_HEADER = 'X-Api-Key';
+
+@Controller('/real-time')
+export default class RealTimeReportsController {
+  #apiKeyValidator: ApiKeyValidator;
+  #reportService: MutationTestingReportService;
+  #blobService: RealTimeMutantsBlobService;
+  #orchestrator: MutationtEventServerOrchestrator;
+  #reportValidator: ReportValidator;
+  #config: Configuration;
+
+  constructor(
+    apiKeyValidator: ApiKeyValidator,
+    dataAcces: DataAccess,
+    mutationtEventServerOrchestrator: MutationtEventServerOrchestrator,
+    reportValidator: ReportValidator,
+    config: Configuration
+  ) {
+    this.#apiKeyValidator = apiKeyValidator;
+    this.#reportService = dataAcces.mutationTestingReportService;
+    this.#blobService = dataAcces.blobService;
+    this.#orchestrator = mutationtEventServerOrchestrator;
+    this.#reportValidator = reportValidator;
+    this.#config = config;
+  }
+
+  @Get('/*')
+  public async getSseEndpointForProject(
+    @Req() req: Request,
+    @Res() res: Response,
+    @QueryParams('module') moduleName: string | undefined
+  ) {
+    const { project, version } = Slug.parse(req.path);
+    const id = {
+      projectName: project,
+      version,
+      moduleName,
+      realTime: true,
+    };
+    const report = await this.#reportService.findOne(id);
+    if (report === null) {
+      throw new NotFound(
+        `Version "${version}" does not exist for "${project}".`
+      );
+    }
+
+    const data = await this.#blobService.getEvents(id);
+    const server = this.#orchestrator.getSseInstanceForProject(id);
+    server.attach(res);
+    data.forEach((mutant) => server.sendMutantTested(mutant));
+  }
+
+  @Delete('/*')
+  public async finish(
+    @Req() req: Request,
+    @QueryParams('module') moduleName: string | undefined,
+    @HeaderParams(API_KEY_HEADER) authorizationHeader: string | undefined
+  ) {
+    if (!authorizationHeader) {
+      throw new Unauthorized(`Provide an "${API_KEY_HEADER}" header`);
+    }
+
+    const { project, version } = Slug.parse(req.path);
+    await this.#apiKeyValidator.validateApiKey(authorizationHeader, project);
+
+    const id = {
+      projectName: project,
+      version: version,
+      moduleName,
+      realTime: true,
+    };
+    const server = this.#orchestrator.getSseInstanceForProject(id);
+    server.sendFinished();
+
+    this.#blobService.delete(id);
+    this.#reportService.delete(id);
```

Comment:
Does this mean it's up to the client to send the completed report again after? This needs to be very explicitly documented. Did we discuss this?

Reply:
I think we did, as the dashboard should work in a similar way as the reference implementation (Stryker.NET)

Reply:
It should be pretty easy to submit the final report from the dashboard backend, right? Could you give me an estimate of the work required?

Reply:
We could maybe add a `HTTP POST /api/real-time/finalize` call, that promotes the real-time results to the finished results.

Maybe add that in a separate PR thought.

Reply:
I think we should put that in a seperate PR (might also be a nice issue for hacktoberfest 😄)